Таратухин Сергей

taratuhin opened the pull request.

taratuhin committed:
Первая реализация стека.

czertyaka commented on cstack.c:
```diff
@@ -1,42 +1,354 @@
+#include  <stdlib.h>
+#include  <string.h>
+
+#define  UNUSED( VAR )  ( void ) ( VAR )
```
В этом макросе наверное смысла больше нет, можно и удалить)

taratuhin replied:
Согласен, убрал.

czertyaka commented on cstack.c:
```diff
@@ -1,42 +1,354 @@
+            node->next = NULL;
+			node->prev = NULL;
+
+			if ( ( list_stacks->tail == NULL ) && ( list_stacks->head == NULL ) )
```
Это условие всегда будет истинно, так как выше вы присваиваете этим полям `NULL`:
https://github.com/taratuhin/prosoft-c-stack/blob/fa055b503375cd5db5f8d6ce8f7d2d4ff76eeeb4/cstack.c#L82-L83

taratuhin replied:
Согласен, убрал.

czertyaka commented on cstack.h:
```diff
@@ -1,13 +1,52 @@
+#include  <stdint.h>
+
+
+typedef  int  hstack_t;
```
Заголовочный файл в задании менять не требуется) Все, что в заголовочном файле, рассматривается как интерфейс библиотеки, а мы в нашей либе хотим предоставить только набор функций и все. Реализация стека должна быть скрыта от пользователя. Да и сделать с этими структурами пользователь все равно ничего не сможет, так как работает с идентификаторами (`int`).

taratuhin replied:
Хотел привести к одному виду и чтоб не было зависимости от операционной системы. Интерфейс вернул.

czertyaka commented on cstack.c:
```diff
@@ -1,42 +1,354 @@
+
+		    if ( node )
+		    {
+                node->hstack = list_stacks->head->hstack + 1;
```
По возможности прошу отступы поправить) Читать будет легче

taratuhin replied:
Хорошо понимаю, поправил.

czertyaka commented on cstack.c:
```diff
@@ -1,42 +1,354 @@
+                node->next = NULL;
+			    node->prev = NULL;
+
+			    if ( ( list_stacks->tail == NULL ) && ( list_stacks->head == NULL ) )
```
Если это условие ложно, то надо освобождать `node`, иначе получим утечку памяти. Более подробно в выводе статического анализатора clang-tidy:
[clang-tidy.log](https://github.com/taratuhin/prosoft-c-stack/files/13624548/clang-tidy.log)

taratuhin replied:
Освободил.

czertyaka commented on cstack.c:
```diff
@@ -1,42 +1,354 @@
+					}
+
+                    element->size_element = size;
+                    element->data = ( void  * ) malloc( size );
```
`malloc` и так возвращает `void*`, приведение тут избыточно

taratuhin replied:
Согласен. Извиняюсь, что так много недочетов, когда писал, болел.

czertyaka reviewed (changes requested):
Есть небольшие замечания)

taratuhin committed:
Добавил освобождение памяти, чтоб не произошло утечки памяти.
Вернул интерфейс в исходное состояние.
Удалил неиспользуемый код.
Выполнил форматирование кода.

taratuhin committed:
Выполнил форматирование кода.
